tests/test_lock.py

```diff
@@ -0,0 +1,284 @@
+import os
+import shutil
+import time
+
+import pandas as pd
+import pytest
+
+import audbackend
+import audeer
+import audformat.testing
+import audiofile
+
+import audb
+
+
+class SlowFileSystem(audbackend.FileSystem):
+    r"""Emulate a slow file system.
+
+    Introduces a short delay when getting a file from the backend.
+    This ensures that timeouts are reached in the tests.
+
+    """
+    def _get_file(self, *args):
+        t = time.time()
+        time.sleep(1)
+        print(f'\nsleep={time.time() - t}')
+        super()._get_file(*args)
+
+
+audbackend.register(
+    'slow-file-system',
+    SlowFileSystem,
+)
+
+
+os.environ['AUDB_CACHE_ROOT'] = pytest.CACHE_ROOT
+os.environ['AUDB_SHARED_CACHE_ROOT'] = pytest.SHARED_CACHE_ROOT
+audb.config.REPOSITORIES = [
+    audb.Repository(
+        name=pytest.REPOSITORY_NAME,
+        host=pytest.FILE_SYSTEM_HOST,
+        backend='slow-file-system',
+    ),
+]
+
+
+DB_NAME = f'test_lock-{pytest.ID}'
+DB_ROOT = os.path.join(pytest.ROOT, 'db')
+DB_VERSION = '1.0.0'
+
+
+def clear_root(root: str):
+    audeer.rmdir(root)
+
+
+@pytest.fixture(
+    scope='function',
+    autouse=True,
+)
+def fixture_remove_db_from_cache():
+    root = audeer.path(pytest.CACHE_ROOT, DB_NAME)
+    clear_root(root)
+
+
+@pytest.fixture(
+    scope='module',
+    autouse=True,
+)
+def fixture_publish_db():
+
+    clear_root(DB_ROOT)
+    clear_root(pytest.FILE_SYSTEM_HOST)
+
+    # create db
+
+    db = audformat.testing.create_db(minimal=True)
+    db.name = DB_NAME
+    db.schemes['scheme'] = audformat.Scheme()
+    audformat.testing.add_table(
+        db,
+        'table',
+        'filewise',
+        num_files=[0, 1, 2],
+    )
+    db.save(DB_ROOT)
+    audformat.testing.create_audio_files(db)
+
+    # publish 1.0.0
+
+    audb.publish(
+        DB_ROOT,
+        DB_VERSION,
+        pytest.PUBLISH_REPOSITORY,
+        verbose=False,
+    )
+
+    yield
+
+    clear_root(DB_ROOT)
+    clear_root(pytest.FILE_SYSTEM_HOST)
+
+
+def load_db(timeout):
+    return audb.load(
```

Comment:
Can you add the following for debugging:

```suggestion
    print(audb.repository(DB_NAME, DB_VERSION))
    return audb.load(
```

Reply:
My guess is that for whatever reason it is not using the correct repo in the runner

Reply:
Ok you are right it's indeed not the slow file system:

```
Repository('data-unittests-local', '/home/runner/work/audb/audb/tests/tmp/repo', 'file-system')
```

and it happens if we run other tests before. Maybe a bug?

Reply:
If I in addition print `audb.config.REPOSITORIES`, it also shows:

```
[Repository('data-unittests-local', '/home/runner/work/audb/audb/tests/tmp/repo', 'file-system')]
```

so it seems that the statement at the top of the test does not have an effect:

```
audb.config.REPOSITORIES = [
    audb.Repository(
        name=pytest.REPOSITORY_NAME,
        host=pytest.FILE_SYSTEM_HOST,
        backend='slow-file-system',
    ),
]
```

Reply:
Maybe you need to provide it as a fixture?

Reply:
But still strange why it is not simply working

Reply:
In other tests we also just set it:

https://github.com/audeering/audb/blob/master/tests/test_info.py#L13-L15

Reply:
Yes, we set it in several tests. Maybe maybe it actually does not have an effect unless you run only this specific test file. The reason we never noticed might be that the very first test we have in `test_api.py` also sets it, but within a test function:

```python
def test_available():

    # Non existing repo
    name = 'non-existing-repo'
    audb.config.REPOSITORIES = [
        audb.Repository(
            name=name,
            host=pytest.FILE_SYSTEM_HOST,
            backend=pytest.BACKEND,
        )
    ]

   ...
```

Reply:
Ok, no that's of course not an explanation since this is actually a non-existing repository we test here :)

Reply:
I fixed it in all tests where wet set `audb.config.REPOSITORIES`.